cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -250,14 +270,14 @@ public KafkaBrokerConfigurationBuilder withListeners(
         List<String> advertisedListeners = new ArrayList<>();
         List<String> securityProtocol = new ArrayList<>();
 
-        boolean isKraftControllerOnly = useKRaft && node.controller() && !node.broker();
-        boolean isKraftBrokerOnly = useKRaft && node.broker() && !node.controller();
+        boolean isKraftControllerOnly = useKRaft && kafkaMetadataConfigState.isKRaftInConfiguration(node) && node.controller() && !node.broker();
 
-        // Control Plane listener => ZooKeeper based brokers and KRaft controllers (including mixed nodes). But not on KRaft broker only nodes
-        if (!isKraftBrokerOnly) {
+        // Control Plane listener set for any KRaft controller, only or mixed, and broker in ZooKeeper-based or migration but NOT when full KRaft
```

Comment:
```suggestion
        // Control Plane listener is set for pure KRaft controller or combined node, and broker in ZooKeeper mode or in migration state. However, it will NOT be set when in full KRaft mode. 
```